lib/em-hiredis/redis_client.rb

```diff
@@ -26,7 +26,8 @@ def initialize(
         uri,
         inactivity_trigger_secs = nil,
         inactivity_response_timeout = nil,
-        em = EventMachine)
+        em = EventMachine,
+        options = {})
```

Comment:
```suggestion
        reconnect_attempts = nil)
```

I would be consistent with other arguments here. `inactivity_trigger_secs` and `inactivity_response_timeout` are both "options", yet are taken regular method parameters.

Reply:
The same goes for `ConnectionManager` parameters. I would define its argument list as `(connection_factory, em = EM, reconnect_attempts = nil)`.

Reply:
Make sense. Although I'm going to go with [positional arguments](https://thoughtbot.com/blog/ruby-2-keyword-arguments) then. I think, right now we have too many parameters in the function declaration. What do you think?

Reply:
I think by "positional arguments" you mean "keyword arguments"?

I much prefer keyword arguments, but in this case positional arguments are appealing because it they don't introduce a breaking change. However, since it's highly unlikely anyone else is actually using this library, I don't think that's such a big concern. So I think it is reasonable to switch to keyword arguments.

However I think if we are going to switch to keyword arguments, we should apply the change consistently across the repo. I think this should be done in a separate PR as it would be a distraction if it was combined with this one.

Reply:
Yes, I did mean keyword arguments :). Make sense what you say.